Modules/Server/Sources/PocketCastsServer/Public/API/ApiServerHandler+Account.swift

```diff
@@ -124,7 +124,27 @@ public extension ApiServerHandler {
         }
     }
 
-    private class func extractErrorResponse(data: Data?, error: Error? = nil) -> APIError? {
+    internal func obtainToken(request: URLRequest, completion: @escaping (Result<AuthenticationResponse, APIError>) -> Void) {
+        urlSession.dataTask(with: request) { data, response, error in
+            guard let responseData = data, error == nil, response?.extractStatusCode() == ServerConstants.HttpConstants.ok else {
+                let errorResponse = ApiServerHandler.extractErrorResponse(data: data, error: error)
+                FileLog.shared.addMessage("Unable to obtain token, status code: \(response?.extractStatusCode() ?? -1), server error: \(errorResponse?.rawValue ?? "none")")
+                completion(.failure(errorResponse ?? .UNKNOWN))
+                return
+            }
+
+            do {
+                let response = try Api_UserLoginResponse(serializedData: responseData)
+                completion(.success(AuthenticationResponse(from: response)))
+            } catch {
+                FileLog.shared.addMessage("Error occurred while trying to unpack token request \(error.localizedDescription)")
```

Comment:
Shall we add any mention of Apple SSO in the log message?

Reply:
I would like to [consolidate](https://github.com/Automattic/pocket-casts-ios/pull/388/files/784a9d8e24e038190c824909660d2c06b00fe842#r993796549) the code to prevent duplication because this is copied and pasted from the login flow. Ony the completion block is changed slightly. We could include a log here by passing in a source but perhaps a log at the start of the Apple SSO attempts would provide the context.